System/Process/Internals.hs

```diff
@@ -414,6 +429,8 @@ foreign import ccall unsafe "runInteractiveProcess"
         -> Ptr FD
         -> Ptr FD
         -> Ptr FD
+        -> Ptr CGid
```

Comment:
I agree that signedness will vary by platform. Do you have a reference that says `-1` is reserved? My concern was that on an unsigned system, `-1` could be interpreted as max_int (which although unlikely could be a user's/group's id). 

I suppose another option would be to add another mask to the flag indicating whether the set the user/group or not. Maybe this is the safest approach.